hs-abci-examples/nameservice/test/Nameservice/Test/E2ESpec.hs

```diff
@@ -65,7 +65,7 @@ spec = do
 
       it "Can query account balances" $ do
         let queryReq = defaultReqWithData addr1
-        ClientResponse{clientResponseData = foundAmount} <- runRPC $ getBalance queryReq
+        ClientResponse{clientResponseData = Just foundAmount} <- runRPC $ getBalance queryReq
```

Comment:
You should be checking that the response code is `0` before asserting `Just` here. Otherwise you will get an uninformative runtime error message. Similar witht he other tests.